### PR TITLE
community[minor]: Integration for `Friendli` LLM and `ChatFriendli` ChatModel.

### PR DESCRIPTION
**Pull request by seuha516**
## Description
- Add [Friendli](https://friendli.ai/) integration for `Friendli` LLM and `ChatFriendli` chat model.
  - Similar python langchain integration is merged in https://github.com/langchain-ai/langchain/pull/17913.
- Documentations corresponding to this change are added.

## Twitter handle
- https://twitter.com/friendliai

### CODE REVIEW

File: libs/langchain-community/src/llms/friendli.ts
Comment (andrewnguonly) [inline libs/langchain-community/src/llms/friendli.ts:83]: For consistency, let's set this to `llama-2-13b-chat` to match `ChatFriendli`.

File: libs/langchain-community/src/chat_models/friendli.ts
Comment (andrewnguonly) [inline libs/langchain-community/src/chat_models/friendli.ts:329]: This needs to be `Accept: "text/event-stream"`.

File: libs/langchain-community/src/chat_models/friendli.ts
Comment (andrewnguonly) [inline libs/langchain-community/src/chat_models/friendli.ts:78]: This is probably a bug in the Friendli API docs, but the example response shows that `role` isn't always returned in the streaming response. If this is true, then this will cause an error.

I'm looking at this documentation: https://docs.friendli.ai/openapi/create-chat-completions#responses

**Review by andrewnguonly — CHANGES_REQUESTED** [review]
Nice job @seuha516. I have a few comments and questions before I approve. Please take a look. Thank you!

File: langchain/src/chat_models/friendli.ts
Comment (jacoblee93) [reply langchain/src/chat_models/friendli.ts:6]: Yes, we should remove as we are planning to completely remove `@langchain/community` from `langchain` in the near future.

File: libs/langchain-community/src/load/import_constants.ts
Comment (jacoblee93) [reply libs/langchain-community/src/load/import_constants.ts:23]: This was likely autogenerated by `yarn build` so I think it's ok!

File: libs/langchain-community/src/llms/friendli.ts
Comment (jacoblee93) [reply libs/langchain-community/src/llms/friendli.ts:83]: I think we do want the `instruct` here actually since this is a text completion endpoint.

I'm also ok to remove the default entirely - we've been generally moving in that direction.

File: libs/langchain-community/langchain.config.js
Comment (jacoblee93) [inline libs/langchain-community/langchain.config.js:317]: You don't need to add your models here since they don't use any peer dependencies.

File: libs/langchain-community/src/llms/friendli.ts
Comment (andrewnguonly) [reply libs/langchain-community/src/llms/friendli.ts:83]: I think we need a default since `model` is required by the Friendli API.

File: libs/langchain-community/langchain.config.js
Comment (seuha516) [reply libs/langchain-community/langchain.config.js]: I removed "llms/friendli" and "chat_models/friendli" from `requiresOptionalDependency`.

File: langchain/src/chat_models/friendli.ts
Comment (seuha516) [reply langchain/src/chat_models/friendli.ts]: I removed `langchain/src/llms/friendli.ts` and `langchain/src/chat_models/friendli.ts` files.

File: libs/langchain-community/src/chat_models/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/chat_models/friendli.ts]: Fixed to `Accept: "text/event-stream"`.

File: libs/langchain-community/src/llms/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/llms/friendli.ts]: Yes, it should be changed.
Fixed to `Accept: "text/event-stream"`, thank you.

File: libs/langchain-community/src/chat_models/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/chat_models/friendli.ts]: I added defaultValue("assistant") for `role`.

File: libs/langchain-community/src/llms/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/llms/friendli.ts]: Updated `Friendli` to return generation metadata when event is completed.

File: libs/langchain-community/src/llms/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/llms/friendli.ts:157]: `X-Friendli-Team` is optional header, I fixed it not to throw `Error` when instantiating without specifying the team.

File: libs/langchain-community/src/llms/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/llms/friendli.ts]: I added `baseUrl` field in `Friendli` and `ChatFriendli` classes.
Now supports Dedicated endpoints with `https://inference.friendli.ai/dedicated`. (`https://inference.friendli.ai/dedicated/{model_id}` is deprecated, `model_id` can be send into `model` field)

File: libs/langchain-community/src/llms/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/llms/friendli.ts:221]: I merged duplicated calling logics and fixed `Friendli` and `ChatFriendli` to get `streaming: boolean` field.

File: libs/langchain-community/src/llms/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/llms/friendli.ts:81]: I added `modelKwargs` field in `Friendli` and `ChatFriendli` to accept new parameters, thank you.

File: libs/langchain-community/src/llms/friendli.ts
Comment (seuha516) [reply libs/langchain-community/src/llms/friendli.ts:226]: I merged duplicated calling api logics into common function, and parameterized it for `stream: boolean`.

**Comment by seuha516** [comment]
Thank you for your detailed and kind comments! I made the requested changes.

**Review by andrewnguonly — APPROVED** [review]
Nice job with all the of updates! @seuha516 

@jacoblee93, are any of the changes under `/langchain` needed? The original `chat_model` and `llm` classes are removed now. Besides that, everything looks good to me.

**Comment by jacoblee93** [comment]
Ah yes can you revert the changes there @seuha516? You fork doesn't seem to allow maintainer edits. Otherwise LGTM!

**Comment by seuha516** [comment]
@jacoblee93 I reverted changes under `/langchain`, please let me know if there is anything wrong. Thank you!

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**
